src/services/blocks/BlocksService.ts

```diff
@@ -449,67 +503,36 @@ export class BlocksService extends AbstractService {
 				}
 			);
 
-			// The block where the runtime is deployed falsely proclaims it would
-			// be already using the new runtime. This workaround therefore uses the
-			// parent of the parent in order to determine the correct runtime under which
-			// this block was produced.
-			let parentParentHash: Hash;
-			if (block.header.number.toNumber() > 1) {
-				parentParentHash = (await api.rpc.chain.getHeader(parentHash))
-					.parentHash;
-			} else {
-				parentParentHash = parentHash;
-			}
+			const parentParentHash: Hash = await this.getParentParentHash(
+				api,
+				parentHash,
+				block
+			);
 
 			const [version, multiplier] = await Promise.all([
 				api.rpc.state.getRuntimeVersion(parentParentHash),
 				api.query.transactionPayment.nextFeeMultiplier.at(parentHash),
 			]);
 
-			[specName, specVersion] = [
-				version.specName.toString(),
-				version.specVersion.toNumber(),
-			];
-
-			// This `extrinsicBaseWeight` changed from using system.extrinsicBaseWeight => system.blockWeights.perClass.normal.baseExtrinsic
-			// in polkadot v0.8.27 due to this pr: https://github.com/paritytech/substrate/pull/6629 .
-			// TODO https://github.com/paritytech/substrate-api-sidecar/issues/393 .
-			// TODO once https://github.com/polkadot-js/api/issues/2365 is resolved we can use that instead.
-			let extrinsicBaseWeight;
-			if (
+			runtimeDoesNotMatch =
 				specName !== api.runtimeVersion.specName.toString() ||
-				specVersion !== api.runtimeVersion.specVersion.toNumber()
-			) {
-				// We are in a runtime that does **not** match the decorated metadata in the api,
-				// so we must fetch the correct metadata, decorate it and pull out the constant
+				specVersion !== api.runtimeVersion.specVersion.toNumber();
+
+			if (runtimeDoesNotMatch) {
```

Comment:
@emostov I could have put the metadata inside of the `Promise.all` but that being said It would have to be a ternary like [this](https://github.com/paritytech/substrate-api-sidecar/blob/master/src/services/blocks/BlocksService.ts#L86). The problem with that is we don't have any `metadata` to resolve the promise with, and `expandMetaData` explicitly takes type `Metadata` only. Any thoughts?